variable/variable.test.functions

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+. test/helper
+. variable/variable.functions
+
+setup () {
+  true # replace as needed
+}
+
+teardown () {
+  true # replace as needed
+}
+
+#
+# Example testing with assert_status - good for multipart tests because
+# you can add a comment and go off of various exit statuses
+#
+
+test_core_variable_startsWith () {
+  core_variable_startsWith "/etc/path/to/file" "/home/"
+  assert_status 1 $? "variable does not start with prefix"
+
+  core_variable_startsWith "/etc/path/to/file" "/etc/"
+  assert_status 0 $? "variable does start with prefix"
+}
+
+#
+# Examples testing with assert_output - simply makes sure stdout is the same.
+# Stderr is not checked but you can of course use redirections. Not good for
+# multipart testing (see the docs, it's because of the pipelines) but in
+# truth I use these more than I use assert_status.  
+#
+
+test_core_variable_firstCharacter()
+{
+  core_variable_firstCharacter "abc" | assert_output "a"
+}
+
+test_core_variable_lastCharacter()
+{
+  core_variable_lastCharacter "abc" | assert_output "c"
+}
+
+#
+# The other useful thing ts provides is $ts_test_dir - the path to a test-
+# specific directory  where you can make files, nearby where some ts-related
+# files are placed in `./tmp`.  The  files are left after tests complete so you
+# can inspect them, but cleaned up prior to each run.  I don't see a function
+# in this module that could use it rationally, but to illustrate...
+#
+
+test_core_variable_trimWhitespace () {
+  mkdir -p "$ts_test_dir"
+  file_with_whitespace="$ts_test_dir/file"
+  cat > "$file_with_whitespace" <<DOC
+
+  some text
+
+DOC
+
+  core_variable_trimWhitespace "$(cat "$file_with_whitespace")" | assert_output "some text"
```

Comment:
cool. So could we invert the logic, and have `assert_*` first, eg
```bash
test_assert_output "some text" core_variable_trimWhitespace "$(cat "$file_with_whitespace")"
``` 
Much neater and no sub shell.